### ISSUE TITLE
linter: effect-exhaustive-dependencies doesn't detect variables used in try finally

### ISSUE THREAD

**Issue by TheNumberOne** [oxc-project/oxc#26093]
### What version of Oxlint are you using?

1.79.0

### What command did you run?

`oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["react"],
  "rules": {
    "react/exhaustive-effect-dependencies": "error"
  }
}
```

### What happened?

If you have an effect and it uses a variable in a `try finally`, the `react/exhaustive-effect-dependencies` will output a false positive that the variable isn't actually used in the effect and is an extra dependency. However, it is used.

It seems to be required that the variable has a dependency on something that changes per render like a `useState` or similar.

```ts
import { useEffect, useState } from 'react';

declare function bb(x: unknown): void

export function Foo() {
  const [foo] = useState('bar');

  useEffect(() => {
    try {
      bb(foo);
    } finally {}
  }, [foo]);
}
```

**Comment by Borjampm** [comment]
Still present in 1.81.0, and the same HIR lowering gap also affects `react/memo-dependencies`, not just `exhaustive-effect-dependencies`:

```tsx
import { useCallback, useState } from 'react';

declare function report(value: string): void;

export function FinallyOnly() {
  const [foo] = useState('bar');
  const cb = useCallback(() => {
    try {
      report('x');
    } finally {
      report(foo);
    }
  }, [foo]);
  return <button type="button" onClick={cb} />;
}
```

→ `react(memo-dependencies): Found extra memoization dependencies` on `foo`, which is a false positive.

Swapping `finally` for `catch` makes it clean, so the trigger is the catch-less `try`: the `Statement::TryStatement` arm in `build_hir.rs` hits the early `return Ok(())` when `handler` is `None` (tracked as a TODO in #26013), dropping the whole statement — a dependency used only in the `try` *body* of a `try/finally` is lost the same way, not just one in the `finally` clause. `try/catch/finally` with both clauses is handled correctly.

Two notes for anyone reproducing:

- The ESLint-port rule `react-hooks/exhaustive-deps` handles these shapes correctly; only the compiler-backed rules are affected.
- Each case must live in its own component — the compiler bails out per component, so a second problematic shape in the same component masks the diagnostic entirely.

**Comment by nathan-knight** [comment]
`react/void-use-memo` is also affected by this:
```
const value = useMemo(() => {
  try {
    return foo;
  } finally {
    report('z');
  }
}, [foo]);
```
Reports `useMemo() callbacks must return a value`.

It's also worth noting that dependencies used in `finally` blocks are flagged regardless of whether or not there is a `catch` block so `try/catch/finally` clauses aren't handled correctly:
```jsx
export function FinallyWithCatch() {
  const [foo] = useState('bar');
  const cb = useCallback(() => {
    try {
      report('x');
    } catch {
      report('y');
    } finally {
      report(foo);
    }
  }, [foo]);
  return <button type="button" onClick={cb} />;
}
```
This flags `foo`
